### PR TITLE
overlord/snapstate,snap,store: track snap-id in side-info and therefore in state

### PR DESCRIPTION
**Pull request by pedronis**
what to do about states already out in the field?

### CODE REVIEW

**Comment by pedronis** [comment]
retest this please

File: snap/info.go [hunk 1]
```diff
@@ -60,8 +60,8 @@ func MountDir(name string, revision int) string {
 // SideInfo holds snap metadata that is not included in snap.yaml or for which the store is the canonical source.
 // It can be marshalled both as JSON and YAML.
 type SideInfo struct {
-	// XXX likely we want also snap-id
 	OfficialName      string `yaml:"name,omitempty" json:"name,omitempty"`
+	SnapID            string `yaml:"snap-id" json:"snap-id"`
```
Comment (chipaca) [inline snap/info.go]: it's `snap_id` in the store's JSON

File: store/snap_remote_repo_test.go [hunk 2]
```diff
@@ -201,6 +201,7 @@ const MockDetailsJSON = `{
                 "publisher": "Canonical",
                 "ratings_average": 0.0,
                 "revision": 22,
+                "snap_id": "iZvp6HUG9XOQv4vuRQL9MlEgKBgFwsc6",
```
Comment (chipaca) [inline store/snap_remote_repo_test.go]: see?

**Comment by chipaca** [comment]
:+1: except for the noted one-char bug

File: snap/info.go (hunk 1 above)
Comment (chipaca) [inline snap/info.go]: oh wait, this is `snap/info`

File: store/details.go [hunk 3]
```diff
@@ -42,6 +42,7 @@ type snapDetails struct {
 	Publisher       string             `json:"publisher,omitempty"`
 	RatingsAverage  float64            `json:"ratings_average,omitempty"`
 	Revision        int                `json:"revision"`
+	SnapID          string             `json:"snap_id"`
```
Comment (chipaca) [inline store/details.go]: and this one is right

**Comment by chipaca** [comment]
extra :+1: because i'm an idiot

**Comment by mvo5** [comment]
:+1:

**Pull request merged**